### PR TITLE
Support for references without .value

### PR DESCRIPTION
**Pull request by markacianfrani**
https://github.com/amzn/style-dictionary/issues/721 

The issue above does a good job explaining the problem. If you omit the `.value` in your token format (as is the case when using the Figma Tokens plugin and also the W3 spec), SD fails the reference stage and results in [object object]. Looking at the object returned if you have this omission, there should be a .value property that you can just reference.

However, this seems way too simple so I'm wondering what edge cases or scenarios I might be overlooking here? 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Comment by markacianfrani** [comment]
Well, the good news is the test suites are really good and the bad news is....the test suites are really good. 

I see now that this might be way more complicated than I originally thought. It's the formatters that are giving me trouble now (compose/object for starters). 

Back to the drawing board!

File: lib/utils/flattenProperties.js [hunk 1]
```diff
@@ -29,6 +29,15 @@ function flattenProperties(properties, to_ret) {
     if (properties.hasOwnProperty(name)) {
       // TODO: this is a bit fragile and arbitrary to stop when we get to a 'value' property.
       if (_.isPlainObject(properties[name]) && ('value' in properties[name])) {
+
+        /**
+         * If .value is omitted from the input
+         * ie value: "{color.core.purple}" instead of "{color.core.purple.value}"
+         */
+        if (properties[name].value.hasOwnProperty('value')) {
```
Comment (markacianfrani) [inline lib/utils/flattenProperties.js:37]: I don't love putting this here, since this is flattenProperties function is supposed to be as generic as possible.

**Comment by markacianfrani** [comment]
Had to go way farther up the chain for this. Doing it at the format level caused all sorts of headaches. 

I did find another edge case--in the current test suite, if you were to change, color.font.interactive.active to `color.brand.secondary`, it causes all sorts of side effects. Mixing and matching using .value and not using .value is going to be difficult I think. Although, I really don't see a legit use case where you would want to be able to use both at the same time. 

But is this even something we want to solve at the Style Dictionary level? I'm totally okay with closing this and solving it at the Figma Tokens Plugin level instead.

**Comment by JackHowa** [comment]
this looks like it'd be really helpful!

**Review by crypt096 — APPROVED** [review]

**Comment by lukasoppermann** [comment]
Hey, can this be merged? I am working on references for my [figma plugin](https://github.com/lukasoppermann/design-tokens/pull/146) which exports mainly for style dictionary. However I would love to keep it in line with the w3c spec. If thus could be merged I wouldn't need to include some hacky solution that adds `.value`.

@dbanksdesign ?

**Comment by dbanksdesign** [comment]
Hey everyone! Sorry I haven't been responsive on this PR, my team just had a big launch at re:invent last week that took all of my attention. I am taking a closer look at this PR today and I would like to get this merged in this week and hopefully do a release.

**Comment by dbanksdesign** [comment]
I took a look at this and figured it would be easier to make the change because it requires change code in a more core place. @markacianfrani I'll list you as the co-author though to give you credit. Take a look at #746 and thanks for your patience!

**Comment by dbanksdesign** [comment]
Closing this in favor of #746 but we will give you credit in the release notes @markacianfrani!
